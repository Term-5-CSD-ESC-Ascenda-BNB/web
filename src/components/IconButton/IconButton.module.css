.iconButton {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  background: #fff;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);

  transition: background 0.15s;
}

.iconButton:hover {
  background: #eeeeee;
}
