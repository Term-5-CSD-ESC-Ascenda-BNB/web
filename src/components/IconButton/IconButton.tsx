import { UnstyledButton } from '@mantine/core';
import styles from './IconButton.module.css';
import React from 'react';
import type { Icon } from '@tabler/icons-react';

interface IconButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  icon: React.ReactElement<Icon>;
}

export function IconButton({ className, icon, ...props }: IconButtonProps) {
  const combinedClassName = className ? `${styles.iconButton} ${className}` : styles.iconButton;

  return (
    <UnstyledButton className={combinedClassName} {...props}>
      {icon}
    </UnstyledButton>
  );
}
