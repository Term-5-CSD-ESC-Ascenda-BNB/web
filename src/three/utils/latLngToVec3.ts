import * as THREE from 'three';

export function latLngToVec3(lat: number, lng: number, radius: number): THREE.Vector3 {
  const elevation = THREE.MathUtils.degToRad(90 - lat);
  const azimuth = THREE.MathUtils.degToRad(lng) + Math.PI / 2; // Add PI/2 to align with the globe's coordinate system

  const position = new THREE.Vector3().setFromSphericalCoords(radius, elevation, azimuth);

  return position;
}
