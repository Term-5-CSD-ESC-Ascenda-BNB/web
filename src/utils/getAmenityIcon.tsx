import {
  IconWind,
  IconBriefcase,
  IconIroning3,
  IconPlugConnected,
  IconShirt,
  IconScissors,
  IconBath,
  IconUsersGroup,
  IconSwimming,
  IconParking,
  IconBell,
  IconScale,
  IconDeviceTv,
  IconPhone,
  IconWriting,
  IconCoffee,
  IconBed,
  IconBabyCarriage,
  IconTrowel,
  IconShoe,
  IconHelpCircle,
} from '@tabler/icons-react';

export function getAmenityIcon(label: string): React.ReactElement {
  const f = label.toLowerCase();

  if (f.includes('air')) return <IconWind size={16} />;
  if (f.includes('business')) return <IconBriefcase size={16} />;
  if (f.includes('iron')) return <IconIroning3 size={16} />;
  if (f.includes('data') || f.includes('port')) return <IconPlugConnected size={16} />;
  if (f.includes('dry')) return <IconShirt size={16} />;
  if (f.includes('hair')) return <IconScissors size={16} />;
  if (f.includes('meeting')) return <IconUsersGroup size={16} />;
  if (f.includes('pool')) return <IconSwimming size={16} />;
  if (f.includes('garage') || f.includes('parking')) return <IconParking size={16} />;
  if (f.includes('room service')) return <IconBell size={16} />;
  if (f.includes('safe')) return <IconScale size={16} />;
  if (f.includes('tv')) return <IconDeviceTv size={16} />;
  if (f.includes('phone')) return <IconPhone size={16} />;
  if (f.includes('desk') || f.includes('workspace')) return <IconWriting size={16} />;
  if (f.includes('coffee') || f.includes('tea') || f.includes('minibar') || f.includes('kettle'))
    return <IconCoffee size={16} />;
  if (f.includes('bed') || f.includes('bedding') || f.includes('linen'))
    return <IconBed size={16} />;
  if (f.includes('towel')) return <IconTrowel size={16} />;
  if (f.includes('crib') || f.includes('infant')) return <IconBabyCarriage size={16} />;
  if (f.includes('bath') || f.includes('soak')) return <IconBath size={16} />;
  if (f.includes('shower')) return <IconShoe size={16} />;

  return <IconHelpCircle size={16} />;
}
