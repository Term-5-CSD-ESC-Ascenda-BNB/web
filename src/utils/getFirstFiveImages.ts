import type { ImageDetails } from '@/types/Hotel';

/**
 * Extracts the first five image URLs from a hotel object.
 * @param imageDetails - The image details object containing prefix, suffix, and count.
 * @return An array of up to five image URLs.
 */
export function getFirstFiveImages(imageDetails: ImageDetails): string[] {
  if (!imageDetails || !imageDetails.prefix || !imageDetails.suffix) {
    return [];
  }

  const images: string[] = [];
  const count = Math.min(imageDetails.count, 5);

  for (let i = 0; i < count; i++) {
    images.push(`${imageDetails.prefix}${i}${imageDetails.suffix}`);
  }

  return images;
}
