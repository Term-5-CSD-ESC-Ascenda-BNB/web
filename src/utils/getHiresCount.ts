export function getHiresCount(hires_string: string): number {
  // separate the string by commas
  const hiresArray = hires_string
    .split(',')
    // filter out any empty strings
    .filter((value) => value.trim() !== '');
  // return the length of the array
  return hiresArray.length;
}
