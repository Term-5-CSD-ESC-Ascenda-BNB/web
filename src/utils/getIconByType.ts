import React from 'react';
import {
  IconBuildingSkyscraper,
  IconPlane,
  IconBuilding,
  IconBuildingBank,
  IconSchool,
  IconToolsKitchen2,
  IconShoppingBag,
  IconBuildingMonument,
  IconMask,
  IconBuildingStadium,
  IconCards,
  IconGolf,
  IconGlassFullFilled,
  IconBriefcase,
  IconBuildingHospital,
  IconBuildingCommunity,
  IconSkiJumping,
  IconAnchor,
  IconDirectionsFilled,
  IconTree,
  IconIceCream,
  IconSunglasses,
  IconBuildingEstate,
  IconMapPin,
  type IconProps,
} from '@tabler/icons-react';

/**
 * Returns the appropriate Tabler icon component for a destination type
 * @param type - The destination type (e.g., 'city', 'airport', 'museum')
 * @returns React component for the icon
 */
export const getIconByType = (type: string): React.FC<IconProps> => {
  const iconMap: Record<string, React.FC<IconProps>> = {
    city: IconBuildingSkyscraper,
    airport: IconPlane,
    hotel: IconBuilding,
    museum: IconBuildingBank,
    museums: IconBuildingBank,
    school: IconSchool,
    restaurant: IconToolsKitchen2,
    shopping: IconShoppingBag,
    historic: IconBuildingMonument,
    monument: IconBuildingMonument,
    theater: IconMask,
    stadium: IconBuildingStadium,
    casino: IconCards,
    golf: IconGolf,
    winery: IconGlassFullFilled,
    business: IconBriefcase,
    medical: IconBuildingHospital,
    civic: IconBuildingCommunity,
    skiing: IconSkiJumping,
    anchor: IconAnchor,
    sign: IconDirectionsFilled,
    tree: IconTree,
    icecream: IconIceCream,
    sunglass: IconSunglasses,
    district: IconBuildingEstate,
  };
  return iconMap[type.toLowerCase()] || IconMapPin;
};
