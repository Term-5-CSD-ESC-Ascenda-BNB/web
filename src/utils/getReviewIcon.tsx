import {
  IconMapPin,
  IconBed,
  IconWifi,
  IconCoffee,
  IconToolsKitchen2,
  IconArmchair,
  IconHeartHandshake,
  IconMoodSmile,
  IconHelpCircle,
  IconUser,
  IconUsersGroup,
  IconBriefcase,
  IconHeart,
} from '@tabler/icons-react';
import React from 'react';

export function getReviewIcon(label: string | undefined): React.ReactElement {
  if (!label) return <IconHelpCircle size={16} />;

  const l = label.toLowerCase();

  // Category scores
  if (l.includes('location')) return <IconMapPin size={16} />;
  if (l.includes('room')) return <IconBed size={16} />;
  if (l.includes('wifi') || l.includes('internet')) return <IconWifi size={16} />;
  if (l.includes('breakfast')) return <IconCoffee size={16} />;
  if (l.includes('food') || l.includes('dining')) return <IconToolsKitchen2 size={16} />;
  if (l.includes('comfort')) return <IconArmchair size={16} />;
  if (l.includes('service')) return <IconHeartHandshake size={16} />;
  if (l.includes('vibe')) return <IconMoodSmile size={16} />;

  // Traveler types
  if (l.includes('solo')) return <IconUser size={16} />;
  if (l.includes('couple')) return <IconHeart size={16} />;
  if (l.includes('family')) return <IconUsersGroup size={16} />;
  if (l.includes('business')) return <IconBriefcase size={16} />;

  return <IconHelpCircle size={16} />;
}
