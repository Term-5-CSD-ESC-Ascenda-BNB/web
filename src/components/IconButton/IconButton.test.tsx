import { render, screen } from '@/tests/utils';
import { describe, it, expect } from 'vitest';
import { IconButton } from './IconButton';
import { IconHome } from '@tabler/icons-react';

describe('IconButton', () => {
  it('renders without crashing and matches snapshot', () => {
    const { container } = render(<IconButton icon={<IconHome />} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the provided icon', () => {
    render(<IconButton icon={<IconHome data-testid="icon" />} />);
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });

  it('applies additional className', () => {
    render(<IconButton icon={<IconHome />} className="extra-class" />);
    expect(screen.getByRole('button')).toHaveClass('extra-class');
  });
});
