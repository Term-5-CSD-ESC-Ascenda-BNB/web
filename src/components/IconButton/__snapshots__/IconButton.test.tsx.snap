// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton > renders without crashing and matches snapshot 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: DM Sans, sans-serif;--mantine-font-family-headings: Libre Franklin, serif;--mantine-primary-color-filled: var(--mantine-color-primary-filled);--mantine-primary-color-filled-hover: var(--mantine-color-primary-filled-hover);--mantine-primary-color-light: var(--mantine-color-primary-light);--mantine-primary-color-light-hover: var(--mantine-color-primary-light-hover);--mantine-primary-color-light-color: var(--mantine-color-primary-light-color);--mantine-primary-color-0: var(--mantine-color-primary-0);--mantine-primary-color-1: var(--mantine-color-primary-1);--mantine-primary-color-2: var(--mantine-color-primary-2);--mantine-primary-color-3: var(--mantine-color-primary-3);--mantine-primary-color-4: var(--mantine-color-primary-4);--mantine-primary-color-5: var(--mantine-color-primary-5);--mantine-primary-color-6: var(--mantine-color-primary-6);--mantine-primary-color-7: var(--mantine-color-primary-7);--mantine-primary-color-8: var(--mantine-color-primary-8);--mantine-primary-color-9: var(--mantine-color-primary-9);--mantine-color-primary-0: #F5F6F9;--mantine-color-primary-1: #E8EAF1;--mantine-color-primary-2: #BAC2D6;--mantine-color-primary-3: #99A4C1;--mantine-color-primary-4: #808AB1;--mantine-color-primary-5: #6E74A2;--mantine-color-primary-6: #626693;--mantine-color-primary-7: #545579;--mantine-color-primary-8: #43455E;--mantine-color-primary-9: #2E2E3D;--mantine-color-secondary-0: #F8F6F4;--mantine-color-secondary-1: #DCD6CC;--mantine-color-secondary-2: #C8BEB0;--mantine-color-secondary-3: #AD9D8A;--mantine-color-secondary-4: #9C8773;--mantine-color-secondary-5: #8F7867;--mantine-color-secondary-6: #786256;--mantine-color-secondary-7: #635249;--mantine-color-secondary-8: #51433D;--mantine-color-secondary-9: #2B231F;}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: var(--mantine-color-primary-4);--mantine-color-primary-text: var(--mantine-color-primary-4);--mantine-color-primary-filled: var(--mantine-color-primary-8);--mantine-color-primary-filled-hover: var(--mantine-color-primary-9);--mantine-color-primary-light: rgba(98, 102, 147, 0.15);--mantine-color-primary-light-hover: rgba(98, 102, 147, 0.2);--mantine-color-primary-light-color: var(--mantine-color-primary-3);--mantine-color-primary-outline: var(--mantine-color-primary-4);--mantine-color-primary-outline-hover: rgba(128, 138, 177, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-4);--mantine-color-secondary-filled: var(--mantine-color-secondary-8);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-9);--mantine-color-secondary-light: rgba(120, 98, 86, 0.15);--mantine-color-secondary-light-hover: rgba(120, 98, 86, 0.2);--mantine-color-secondary-light-color: var(--mantine-color-secondary-3);--mantine-color-secondary-outline: var(--mantine-color-secondary-4);--mantine-color-secondary-outline-hover: rgba(156, 135, 115, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: var(--mantine-color-primary-6);--mantine-color-primary-text: var(--mantine-color-primary-filled);--mantine-color-primary-filled: var(--mantine-color-primary-6);--mantine-color-primary-filled-hover: var(--mantine-color-primary-7);--mantine-color-primary-light: rgba(98, 102, 147, 0.1);--mantine-color-primary-light-hover: rgba(98, 102, 147, 0.12);--mantine-color-primary-light-color: var(--mantine-color-primary-6);--mantine-color-primary-outline: var(--mantine-color-primary-6);--mantine-color-primary-outline-hover: rgba(98, 102, 147, 0.05);--mantine-color-secondary-text: var(--mantine-color-secondary-filled);--mantine-color-secondary-filled: var(--mantine-color-secondary-6);--mantine-color-secondary-filled-hover: var(--mantine-color-secondary-7);--mantine-color-secondary-light: rgba(120, 98, 86, 0.1);--mantine-color-secondary-light-hover: rgba(120, 98, 86, 0.12);--mantine-color-secondary-light-color: var(--mantine-color-secondary-6);--mantine-color-secondary-outline: var(--mantine-color-secondary-6);--mantine-color-secondary-outline-hover: rgba(120, 98, 86, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="m_71ac47fc mantine-AspectRatio-root"
    style="--ar-ratio: 1;"
  >
    <button
      class="mantine-focus-auto _iconButton_57bad1 m_87cf2631 mantine-UnstyledButton-root"
      type="button"
    >
      <svg
        class="tabler-icon tabler-icon-home "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12l-2 0l9 -9l9 9l-2 0"
        />
        <path
          d="M5 12v7a2 2 0 0 0 2 2h10a2 2 0 0 0 2 -2v-7"
        />
        <path
          d="M9 21v-6a2 2 0 0 1 2 -2h2a2 2 0 0 1 2 2v6"
        />
      </svg>
    </button>
  </div>
</div>
`;
